### PR TITLE
Adding structure expectations to CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by eddie-knight**
Following the call today, I'm hoping this PR will get the conversation started regarding a standardized organization.

https://github.com/eddie-knight/cloud-service-certification/blob/citihub/proposed-reorganization/docs/directory-structure.md

### CODE REVIEW

File: docs/directory-structure.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# CSC Directory Structure
+
+The CSC repository contains three types of content: Service Accellerator Templates, Infrastructure as Code, and Automated Testing.
```
Comment (mcleo-d) [inline docs/directory-structure.md:3]: Typo : `Accellerator` should be `Accelerator`

File: docs/directory-structure.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+This document is designed to establish a standard layout that each content type should meet when being contributed to the repository.
+
+## Service Accellerator Templates
```
Comment (mcleo-d) [inline docs/directory-structure.md:7]: Typo : `Accellerator` should be `Accelerator`

File: docs/directory-structure.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+**Parent directory:**
+
+1. Cloud service provider name
+1. `service_accellerator_templates/`
```
Comment (mcleo-d) [inline docs/directory-structure.md:12]: Typo : `Accellerator` should be `Accelerator`

File: docs/directory-structure.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+1. Name of topic/category
+
+*Example:*
+`aws/service_accellerator_templates/dynamodb`
```
Comment (mcleo-d) [inline docs/directory-structure.md:16]: Typo : `Accellerator` should be `Accelerator`

File: docs/directory-structure.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+One or more modules will be used within a configuration, which should live within it's own directory that is named based on the topic.
+
+*Example:*
+
+```bash
+gcp/
+    service_accellerator_templates/
+        postgresql/
+            # Service Accellerator Templates
```
Comment (mcleo-d) [inline docs/directory-structure.md:39]: Typo : `Accellerator` should be `Accelerator`

File: docs/directory-structure.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+Terraform contributions should always take advantage of [modules](https://www.terraform.io/docs/language/modules/develop/index.html). Modules should live in `<CSP>/terraform/modules`. Containing all modules for a cloud provider together will help prevent unecessary code duplication.
+
+One or more modules will be used within a configuration, which should live within it's own directory that is named based on the topic.
+
+*Example:*
+
+```bash
+gcp/
+    service_accellerator_templates/
```
Comment (mcleo-d) [inline docs/directory-structure.md:37]: Typo : `service_accellerator_templates/` should be `service_accelerator_templates/`

File: docs/directory-structure.md [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+        postgresql/
+            # Service Accellerator Templates
+        gke/
+            # Service Accellerator Templates
```
Comment (mcleo-d) [inline docs/directory-structure.md:41]: Typo : `Accellerator` should be `Accelerator`

File: docs/directory-structure.md [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+## Infrastructure as Code
+
+**Parent directory:**
+
+1. Cloud service provider name.
+1. Name of IaC format (currently only Terraform is being used)
+
+**Terraform directory organization:**
+
+Terraform contributions should always take advantage of [modules](https://www.terraform.io/docs/language/modules/develop/index.html). Modules should live in `<CSP>/terraform/modules`. Containing all modules for a cloud provider together will help prevent unecessary code duplication.
```
Comment (mcleo-d) [inline docs/directory-structure.md:29]: Typo: `unecessary` should be `unnecessary`

**Comment by mcleo-d** [comment]
Hi @eddie-knight

Thanks so much for the PR 🚀  

I have taken a brief read through of `docs/directory-structure.md` and found a few minor typos that I've highlighted in the commit.

I've also noticed the FINOS CLABot has been triggered. I'll email you directly regarding this issue.

Thanks once again! 😁 

James.

**Comment by mcleo-d** [comment]
Hi @eddie-knight

I have assigned this PR to @peterrhysthomas for review and will add to the next CSC meeting for you to introduce to the group. 

Also, please refer to the following comment regarding the CLABot warning ... https://github.com/finos/cloud-service-certification/pull/104#issuecomment-790697916

Speak soon,

James.

**Comment by eddie-knight** [comment]
Per previous conversations, I've modified this PR to simply state the expectations within the contribution docs.

If this is agreeable, we can follow up with additional PRs that adjust the existing contributions to meet the standard outlined here.

**Comment by mcleo-d** [comment]
Hi @eddie-knight,

Thanks for the comment below. Feel free to add a link to this PR in the comments of https://github.com/finos/cloud-service-certification/issues/112 to make sure we discuss with @peterrhysthomas / @abdullahgarcia and agree a way forward at this weeks mid sprint session.

Keep safe and speak soon 🤘😆 

James.

> Per previous conversations, I've modified this PR to simply state the expectations within the contribution docs.
> 
> If this is agreeable, we can follow up with additional PRs that adjust the existing contributions to meet the standard outlined here.

**Review by abdullahgarcia — CHANGES_REQUESTED** [review]
@eddie-knight , can we please look over the proposed directory structure? 

I'm currently concerned about how we handle the services for each cloud service provider.

We currently hold a sub-directory for each service to ensure it is easy to navigate and keep segregation for the files involved. 

That isn't depicted in the current structure.

Please let me know how we can address this matter.

**Comment by eddie-knight** [comment]
I'd love to discuss and apply any suggestions. Should we discuss this on the call tomorrow or do you want to propose an improvement here via the comments?

**Comment by abdullahgarcia** [comment]
Let's see how the session goes tomorrow. If there's time, we can discuss it and perhaps gain everyone else's feedback too. Otherwise, we can set something up.

File: .github/CONTRIBUTING.md [hunk 9]
```diff
@@ -44,3 +44,81 @@ To make review of PRs easier, please:
...
+1. Terraform Modules
+    * The central logic of all IaC should be reusable and live in a module
+    * Modules should all live in their own subdirectory, with each module in another subdirectory named after the resource it pertains to
+    * Modules should be usable by any config that needs it (agnostic and configurable)
+    * Ideally, a separate module should exist for each type of resource that needs to be created
+  
+#### Example CSP Directory Structure
+
+```
+# finos/cloud-services-certification
```
[the remark this answers is not in the document]
Comment (eddie-knight) [reply .github/CONTRIBUTING.md:91]: Added these changes in my previous commit

[the remark this answers is not in the document]
Comment (eddie-knight) [reply .github/CONTRIBUTING.md:91]: changes applied in commit d416596

**Review by abdullahgarcia — APPROVED** [review]
As per the last CSC call, all looks good.

**Review by peterrhysthomas — APPROVED** [review]

**Comment by mcleo-d** [comment]
Congratulations on the merge @eddie-knight and @peterrhysthomas 🕺🏻 💃🏻  🎸 🎵 🎶 🎵 🎶 🎵 🎶

**Pull request merged**